open_spiel/games/solitaire.h

```diff
@@ -0,0 +1,409 @@
+#ifndef THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+#define THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+
+#include <any>
+#include <array>
+#include <deque>
+#include <memory>
+#include <set>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include "open_spiel/spiel.h"
+#include "open_spiel/abseil-cpp/absl/types/optional.h"
+
+// An implementation of klondike solitaire:
+// https://en.wikipedia.org/wiki/Klondike_(solitaire) More specifically, it is
+// K+ solitaire, which allows the player to play any card from the deck/waste
+// that would normally become playable after some number of draws in standard
+// klondike solitaire. For a more in-depth description of K+ solitaire, see
+// http://web.engr.oregonstate.edu/~afern/papers/solitaire.pdf. This
+// implementation also gives rewards at intermediate states like most electronic
+// versions of solitaire do, rather than only at terminal states.
+
+// ANSI color codes
+inline constexpr const char *kReset = "\033[0m";
+inline constexpr const char *kRed = "\033[31m";
+inline constexpr const char *kBlack = "\033[37m";
+
+// Unicode Glyphs
+inline constexpr const char *kGlyphHidden = "\U0001F0A0";
+inline constexpr const char *kGlyphEmpty = "\U0001F0BF";
+inline constexpr const char *kGlyphSpades = "\U00002660";
+inline constexpr const char *kGlyphHearts = "\U00002665";
+inline constexpr const char *kGlyphClubs = "\U00002663";
+inline constexpr const char *kGlyphDiamonds = "\U00002666";
+inline constexpr const char *kGlyphArrow = "\U00002190";
+
+namespace open_spiel::solitaire {
+
+// Default Game Parameters =====================================================
+
+inline constexpr int kDefaultPlayers = 1;
+inline constexpr int kDefaultDepthLimit = 150;
+inline constexpr bool kDefaultIsColored = false;
+
+// Enumerations ================================================================
```

Comment:
Do these need to be in the header file or can we put them in the .cc in an anonymous namespace?

Reply:
Same as other comments, the rest of solitaire.h depends on these enums so it can't be moved to solitaire.cc.